### PR TITLE
Add pscale v230

### PR DESCRIPTION
**Pull request by jooseppi-luna**
# Description
* Add configuration files to install Powerscale v2.3.0
* Add function to utils to check a minimum version
* Add code to powerscale.go to make sure the driver meets the minimum version

# GitHub Issues
https://github.com/dell/csm/issues/243

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Ran unit tests for both controller and driver, unable to test config files themselves since PowerScale v2.3.0 has not been released yet

### CODE REVIEW

File: controllers/csm_controller.go [hunk 1]
```diff
@@ -91,11 +90,16 @@ const (
...
-var dMutex sync.RWMutex
-var configVersionKey = fmt.Sprintf("%s/%s", MetadataPrefix, "CSIoperatorConfigVersion")
+var (
+	dMutex           sync.RWMutex
+	configVersionKey = fmt.Sprintf("%s/%s", MetadataPrefix, "CSIoperatorConfigVersion")
 
-// StopWatch - watcher stop handle
-var StopWatch = make(chan struct{})
+	// PScaleSupportedDriverVersions is a list of supported Powerscale driver versions
+	PScaleSupportedDriverVersions = map[string]bool{"v2.2.0": true, "v2.3.0": true}
```
[the remark this answers is not in the document]
Comment (jooseppi-luna) [reply controllers/csm_controller.go:98]: Hmmm ok, I understand what you're saying, so do you think that we'd be better off just checking a minimum version instead?

**Review by nb950 — APPROVED** [review]

**Comment by rajendraindukuri** [comment]
Hi @jooseppi-luna , This looks like a release activity. If yes, please link this to github issue https://github.com/dell/csm/issues/243, I have added "area/csm-operator" label for the same. If not, suggest you to create a new github issue.

File: controllers/csm_controller.go [hunk 2]
```diff
@@ -694,17 +698,17 @@ func (r *ContainerStorageModuleReconciler) removeDriver(ctx context.Context, ins
...
-	}
-
 	// Check drivers
 	switch cr.Spec.Driver.CSIDriverType {
 	case csmv1.PowerScale:
 		err := drivers.PrecheckPowerScale(ctx, cr, r.GetClient())
 		if err != nil {
 			return fmt.Errorf("failed powerscale validation: %v", err)
 		}
+		if cr.Spec.Driver.ConfigVersion == "" || !(PScaleSupportedDriverVersions[cr.Spec.Driver.ConfigVersion]) {
```
Comment (medegw01) [inline controllers/csm_controller.go:708]: Please move this version check to https://github.com/dell/csm-operator/blob/7f1d2c2c84c7c6f6a88aaf2e3a134f58759fa565/pkg/drivers/powerscale.go#L27

All module/driver specific checks should be handled in their respective pkg

**Review by medegw01 — CHANGES_REQUESTED** [review]

**Comment by bharathsreekanth** [comment]
Please update with GitHub issue and update the description to indicate what tests were performed. Also, attach any screenshots if relevant.

File: controllers/csm_controller.go (hunk 2 above)
Comment (jooseppi-luna) [reply controllers/csm_controller.go]: @medegw01 done

**Comment by bharathsreekanth** [comment]
> Hi @jooseppi-luna , This looks like a release activity. If yes, please link this to github issue [dell/csm#243](https://github.com/dell/csm/issues/243), I have added "area/csm-operator" label for the same. If not, suggest you to create a new github issue.

Hi @rajendraindukuri , I am not sure if this qualifies as a release activity yet since these files are basically same as those for v2.2. In order for us to support with the powerscale nightly and test driver install and upgrade via operator . we need a version that is not 2.2.0 version, hence the changes IMO. Jooseppi can correct me there. It is still very much under development, and I would still consider it as feature related work rather than release activity. If I am missing anything, please correct me.

**Comment by jooseppi-luna** [comment]
> > Hi @jooseppi-luna , This looks like a release activity. If yes, please link this to github issue [dell/csm#243](https://github.com/dell/csm/issues/243), I have added "area/csm-operator" label for the same. If not, suggest you to create a new github issue.
> 
> Hi @rajendraindukuri , I am not sure if this qualifies as a release activity yet since these files are basically same as those for v2.2. In order for us to support with the powerscale nightly and test driver install and upgrade via operator . we need a version that is not 2.2.0 version, hence the changes IMO. Jooseppi can correct me there. It is still very much under development, and I would still consider it as feature related work rather than release activity. If I am missing anything, please correct me.

I would agree with this @bharathsreekanth

**Review by medegw01 — APPROVED** [review]

**Review by medegw01 — DISMISSED** [review]

**Review by abhi16394 — APPROVED** [review]

**Review by medegw01 — APPROVED** [review]

**Pull request merged**